src/mixins/document-mixin.ts

```diff
@@ -17,14 +19,20 @@ export default class DocumentMixin extends Vue {
     }
   }
 
-  private pdfjsLib: any
-
   created () {
     // NB: we load the lib and worker this way to avoid a memory leak (esp in unit tests)
     // NB: must use require instead of import or this doesn't work
     // NB: must use legacy build for unit tests not running in Node 18+
-    this.pdfjsLib = require('pdfjs-dist/legacy/build/pdf.js')
-    this.pdfjsLib.GlobalWorkerOptions.workerSrc = require('pdfjs-dist/legacy/build/pdf.worker.entry')
+    try {
+      if (typeof window === 'undefined' || !('Worker' in window)) {
+        throw new Error('Web Workers not supported in this environment.')
+      }
+
+      window.pdfjsWorker = pdfjsWorker
+      pdfjs.GlobalWorkerOptions.workerSrc = `https://cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`
```

Comment:
I've spent hours on trying to get the pdfjs library to work with Vite. This is literally the only way that I got it to work. Unit tests pass.

Here's a link if you want to try it: https://business-create-dev--pr-556-nilgmamu.web.app/incorporation-memorandum?id=TA7zV53Hu5

Reply:
I had a few links for this - one sec, have you tried them?

https://github.com/originjs/vite-plugins/tree/main/packages/vite-plugin-commonjs

https://erindoyle.dev/using-pdfjs-with-vite/

Reply:
The second link is exactly what I used to get it to work 😆

Reply:
Rofl sorry HAHA i just looked and saw.

Reply:
```
  const pdfjsWorker = await import('pdfjs-dist/build/pdf.worker.entry');

  pdfjs.GlobalWorkerOptions.workerSrc = pdfjsWorker;
  ```

Reply:
Haven't tried -^

Reply:
Travis, you're incredible. Something of the sorts works. Now, I don't to add the pdfjs library to the CSP anymore which is what I would've preferred! Thank you so much Travis!

Reply:
Travis, please check latest commit. I managed to get pdfjs to work without CDN and with the unit tests to run also.

Reply:
If we do need an external JS library (at runtime) then I recommend we cache it locally (as we've done with address-complete). No CDN.

Reply:
Agreed. Makes sense.